ethcore/src/client/client.rs

```diff
@@ -1535,6 +1499,52 @@ impl Drop for Client {
 	}
 }
 
+fn transaction_receipt(transaction: Option<LocalizedTransaction>, receipts: Option<Vec<Receipt>>) -> Option<LocalizedReceipt> {
+	let tx_and_sender = transaction
+		.and_then(|tx| tx.sender().ok().map(|sender| (tx, sender)));
```

Comment:
This function is not really needed at all, but it is very difficult to test `Client::transaction_receipt` so I wanted to move as much logic as possible here to be able to test it properly.

Also most of the options could be replaced with `.expect("Blockchain database is never pruned and stays consistent.")` but I think current approach with `.and_then` chain is more future proof.

I will move `Options` to `Client::transaction_receipt`